### PR TITLE
add mass flux divergence to detrainment

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Adds a term for mass flux divergence in detrainment. This is not implemented for diagnostic edmf as calculating the mass flux divergence at level k requires mass flux at level k, which we don't have when solving edmf variable at level k. We can get around it by using levels k-1 and k-2 to calculate the divergence, but it's a bit more complicated and we are not focusing on improving diagnostic EDMF now.

I keep both `vert_div` and `massflux_vert_div` term, to have no behavior change for diagnostic edmf. We can reevaluate later if `vert_div` is still needed. I also remove `GeneralizedHarmonicsEntrainment` and `GeneralizedHarmonicsDetrainment`, as it's unlikely we will use them.
 
part of #2705 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: post_processing/ci_plots.jl [hunk 1]
```diff
@@ -954,6 +954,7 @@ EDMFBoxPlots = Union{
     Val{:prognostic_edmfx_gabls_box},
     Val{:prognostic_edmfx_bomex_fixtke_box},
     Val{:prognostic_edmfx_bomex_box},
+    Val{:prognostic_edmfx_bomex_column},
```
Comment (szy21) [inline post_processing/ci_plots.jl:957]: remove this

**Comment by szy21** [comment]
I will update ClimaParams for other environment as well.

**Review by trontrytel — APPROVED** [review]

File: post_processing/ci_plots.jl (hunk 1 above)
Comment (trontrytel) [reply post_processing/ci_plots.jl:957]: do you still want to remove this?

Comment (szy21) [reply post_processing/ci_plots.jl:957]: yes I will remove this

**Pull request merged**